util/vendor.py

```diff
@@ -214,11 +214,28 @@ def import_from_upstream(upstream_path, target_path, exclude_files=[]):
     _cp_from_upstream(upstream_path, target_path, exclude_files)
 
 
-def apply_patch(basedir, patchfile, strip_level=1):
+def apply_patch(basedir, patchfile, strip_level=1, check_patches=False):
     cmd = ['git', 'apply', '-p' + str(strip_level), str(patchfile)]
+
+    if check_patches:
+        # Pass arguments to both check and apply the patches
+        cmd += ['--check', '--apply']
```

Comment:
Is `git apply --check` stricter than `--apply`, i.e. does it fail if patches would be applied with fuzz?

And if so, do we need this additional strict-ness going beyond what the vendor tool normally would do?

Reply:
I'm mainly worried about being more strict in CI than we are in the "normal" use of the tool. If git somehow applies a patch, then the vendor tool won't complain in normal use. Having it only complain in CI will increase frustration for users. So it feels like either add `--strict` always to the tool, or not at all. (And I personally think that we can and should be as lenient as possible here; the vendoring process is already annoying enough for our users, if we can meet them half-way, we should.)